open_instruct/utils.py

```diff
@@ -879,3 +868,74 @@ def clean_last_n_checkpoints(output_dir: str, keep_last_n_checkpoints: int) -> N
         checkpoints = sorted(folders, key=lambda x: int(x.split("_")[-1]))
         if len(checkpoints) > keep_last_n_checkpoints:
             shutil.rmtree(os.path.join(output_dir, checkpoints[0]))
+
+
+# ----------------------------------------------------------------------------
+# Ai2 user utilities
+@dataclass
+class BeakerRuntimeConfig:
+    beaker_workload_id: str
+    beaker_node_hostname: str
+    beaker_experiment_url: str
+
+
+def maybe_get_beaker_config():
+    beaker_runtime_config = None
+    if "BEAKER_JOB_ID" in os.environ:
+        beaker_runtime_config = BeakerRuntimeConfig(
+            beaker_workload_id=os.environ["BEAKER_WORKLOAD_ID"],
+            beaker_node_hostname=os.environ["BEAKER_NODE_HOSTNAME"],
+            beaker_experiment_url=f"https://beaker.org/ex/{os.environ['BEAKER_WORKLOAD_ID']}/",
+        )
+    return beaker_runtime_config
+
+
+def maybe_use_ai2_wandb_entity() -> Optional[str]:
+    """Ai2 internal logic: try use the ai2-llm team if possible. Should not affect external users."""
+    import wandb
+
+    wandb.login()
+    api = wandb.Api()
+    current_user = api.viewer
+    teams = current_user.teams
+    if "ai2-llm" in teams:
+        return "ai2-llm"
+    else:
+        return None
+
+
+def maybe_use_ai2_hf_entity() -> Optional[str]:
+    """Ai2 internal logic: try use the allenai entity if possible. Should not affect external users."""
+    orgs = HfApi().whoami()
+    orgs = [item["name"] for item in orgs["orgs"]]
+    if "allenai" in orgs:
+        return "allenai"
+    else:
+        return None
+
+
+def submit_beaker_eval_jobs(hf_repo_id: str, hf_repo_revision: str) -> None:
+
+    command = f"""
+    python scripts/submit_eval_jobs.py \
+        --model_name hf-{hf_repo_revision} \
+        --hf_revision {hf_repo_revision} \
+        --location {hf_repo_id} \
+        --is_tuned \
+        --workspace tulu-3-results \
+        --preemptible \
+        --use_hf_tokenizer_template \
+        --beaker_image costah/open_instruct_test \
```

Comment:
probably should swap to `nathanl/open_instruct_olmo_auto`, the autobuilt beaker image

Reply:
Why the olmo version over `nathanl/open_instruct_auto`?

Reply:
Yep we should do that once the PR is merged. I need to use the custom image for testing out this workflow.

Reply:
Have just revert back to the auto build image

Reply:
olmo version because that uses the newest vllm, so its more useful at eval time. The benefit of the non-olmo image is that it has deepspeed installed, but that isn't useful at eval time.